### PR TITLE
[Feature] Ore Cluster Item

### PR DESCRIPTION
**Pull request by justinrusso**

### CODE REVIEW

File: src/main/resources/assets/base/models/item/cluster.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (SkySom) [inline src/main/resources/assets/base/models/item/cluster.json]: This file isn't needed

File: src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java [hunk 2]
```diff
@@ -45,7 +45,12 @@ public ItemMaterialPart(MaterialUser materialUser) {
 
     @Override
     public int getColorFromItemstack(@Nonnull ItemStack itemStack, int tintIndex) {
-        return tintIndex == 0 ? this.getMaterialPartFromItemStack(itemStack).getColor() : -1;
+        MaterialPart materialPart = this.getMaterialPartFromItemStack(itemStack);
+        if (materialPart.getPart().hasOverlayTexture()) {
```
Comment (SkySom) [inline src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java:49]: Really makes sense still to just keep tintIndex the same for everything? Unless layers do equal the tintIndex in items?

**Review by SkySom — CHANGES_REQUESTED** [review]

File: src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java (hunk 2 above)
Comment (justinrusso) [reply src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java:49]: Tintindex is the layer from my understanding

File: src/main/resources/assets/base/models/item/cluster.json (hunk 1 above)
Comment (justinrusso) [reply src/main/resources/assets/base/models/item/cluster.json]: Not sure how this got in.. will remove it

File: src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java (hunk 2 above)
Comment (SkySom) [reply src/main/java/com/teamacronymcoders/base/materialsystem/items/ItemMaterialPart.java:49]: Fair enough, as long as it works.

File: src/main/resources/assets/base/models/item/cluster.json (hunk 1 above)
Comment (justinrusso) [reply src/main/resources/assets/base/models/item/cluster.json]: what would I use instead?

File: src/main/java/com/teamacronymcoders/base/materialsystem/parts/PartBuilder.java [hunk 3]
```diff
@@ -44,9 +45,14 @@ public PartBuilder setAdditionalOreDictNames(String... additionalOreDictNames) {
...
 
+    public PartBuilder setOverlay(boolean overlay) {
+        this.overlay = overlay;
+        return this;
+    }
+
     public Part build() throws MaterialException {
         validate();
-        Part part =  new Part(name, oreDictName, partType, ownerId, additionalOreDictNames);
+        Part part =  new Part(name, oreDictName, partType, ownerId, additionalOreDictNames, overlay);
```
Comment (SkySom) [inline src/main/java/com/teamacronymcoders/base/materialsystem/parts/PartBuilder.java]: Rogue extra space

**Review by SkySom — CHANGES_REQUESTED** [review]

**Review by SkySom — APPROVED** [review]

**Pull request merged**